### PR TITLE
Update Plan ThankYou/Congrats page

### PR DESCRIPTION
**Pull request by candy02058912**
Fixes https://github.com/Automattic/dotcom-forge/issues/2900

## Proposed Changes

* Update plans thank you page to redesign V2

| Before  | After |
| :-------------: | :-------------: |
|   | |

Redesign v2 Mobile
mobile

## Testing Instructions

* Select a WordPress.com plan and purchase it
* Upsells should still work, you can click on "No thanks" and it should show this ThankYou page
* Check the ThankYou page for mobile/desktop
* Test around different purchases (email/Google/etc.) try to break it
* See if tracks events are triggered

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] ~[Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?~
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] ~For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/header.jsx [hunk 1]
```diff
@@ -56,47 +57,14 @@ export class CheckoutThankYouHeader extends PureComponent {
...
 		upgradeIntent: PropTypes.string,
+		isRedesignV2: PropTypes.bool,
 	};
 
 	isSearch() {
 		const { purchases } = this.props;
 		return purchases?.length > 0 && purchases[ 0 ].productType === 'search';
 	}
 
-	getHeading() {
```
Comment (candy02058912) [inline client/my-sites/checkout/checkout-thank-you/header.jsx:66]: Moved this to [get-heading.ts](https://github.com/Automattic/wp-calypso/pull/78931/files#diff-28daa95864f94a5364e44ec5e03d5d2f5fbdec7ec13def80ea0cafebde2ab8dd)

File: client/my-sites/checkout/checkout-thank-you/index.tsx [hunk 2]
```diff
@@ -275,6 +280,27 @@ export class CheckoutThankYou extends Component<
 		}
 	}
 
+	/**
+	 * Determines whether the current checkout flow is for a redesign V2 purchase.
+	 * Used for gradually rolling out the redesign.
+	 *
+	 * @returns {boolean} True if the checkout flow is for a redesign V2 purchase, false otherwise.
+	 */
+	isRedesignV2 = () => {
```
Comment (candy02058912) [inline client/my-sites/checkout/checkout-thank-you/index.tsx:291]: Will move this out to a separate file in another PR so we can write tests for this.

File: client/my-sites/checkout/checkout-thank-you/index.tsx [hunk 3]
```diff
@@ -275,6 +280,27 @@ export class CheckoutThankYou extends Component<
...
+		// Fallback to old design when there is a failed purchase.
+		const failedPurchases = getFailedPurchases( this.props );
+		if ( failedPurchases.length > 0 ) {
+			return false;
+		}
+
+		// ThankYou page for only purchasing a plan.
+		const purchases = getPurchases( this.props );
+		if ( purchases.length === 1 ) {
+			return isWpComPlan( purchases[ 0 ].productSlug );
```
Comment (candy02058912) [inline client/my-sites/checkout/checkout-thank-you/index.tsx:301]: We cannot use `isPlan` because it includes Jetpack plans, which we are not supporting. They have a different setup flow.

**Review by lupus2k — APPROVED** [review]
It works great, you did an amazing job, @candy02058912 🚀

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/default-contact.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
```
Comment (paulopmt1) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/default-contact.tsx:1]: This file [looks pretty close to this one](https://github.com/Automattic/wp-calypso/blob/update/plans-congrats/client/my-sites/checkout/checkout-thank-you/marketplace/masterbar-styled/default-contact.tsx). Do we have a special reason to not reuse/extend that one, or is the goal here to isolate everything by rewriting the components?

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import { Global, css } from '@emotion/react';
```
Comment (paulopmt1) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx:1]: I see, it seems we want to rewrite it since this file is also [close to this one](https://github.com/Automattic/wp-calypso/blob/update/plans-congrats/client/my-sites/checkout/checkout-thank-you/marketplace/masterbar-styled/index.tsx).

Will we remove those reference files in a follow-up PR after ensuring this one works as expected?

File: client/my-sites/checkout/checkout-thank-you/style.scss [hunk 6]
```diff
@@ -983,3 +983,200 @@ main.akismet-checkout-thank-you {
 .jetpack-instruction-list__item .gridicon {
 	margin-bottom: -3px;
 }
+
+// Design i2 overrides, the purchase-detail component is also used by the /plans/my-plan page to render plan features.
```
Comment (paulopmt1) [inline client/my-sites/checkout/checkout-thank-you/style.scss]: Since we want to override it and have everything isolated, should we move this code to a new file inside `redesign-v2` folder as well?

File: client/my-sites/checkout/checkout-thank-you/index.tsx (hunk 2 above)
Comment (paulopmt1) [reply client/my-sites/checkout/checkout-thank-you/index.tsx:291]: Nice, that sounds good. Will it also lives inside `redesign-v2` folder?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8353915

Thank you @candy02058912 for including a screenshot in the description! This is really helpful for our translators.

**Review by paulopmt1 — APPROVED** [review]
Hey @candy02058912, plugins, themes, and plans are working as expected, good job!

- I [added minor changes here](https://github.com/Automattic/wp-calypso/pull/78931/commits/8c2993da05b7c8fe54442c905ebd6d0d5e5e95f0), please check if it makes sense to you.

- I also added string freeze since most of the strings doesn't have translations for me:

	image


- Other than that, we can handle my comments/questions as follow-ups.

**Review by JessBoctor — COMMENTED** [review]
The redesign is looking really nice. 

I ran into one small issue where the confetti animation was shown before the checkout page had finished loading. 

GH78931-UpgradeThankYouReview-2023-07-06 12 00 29

I checked using both Store Sandbox ( testoctor) and Production ( jessboctor ) and it happened both times. I was using Chrome on macOS.

Let know if you need more information. Otherwise, this is looking really great!

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/default-contact.tsx (hunk 4 above)
Comment (candy02058912) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/default-contact.tsx:1]: This will be the main component that we will use. it will be that the marketplace one (old one) should link to this one in another PR.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx (hunk 5 above)
Comment (candy02058912) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx:1]: > Will we remove those reference files in a follow-up PR after ensuring this one works as expected?

Yeah we will try to keep it down to one file in a future PR.
Issue tracking this :https://github.com/Automattic/dotcom-forge/issues/2963

**Comment by candy02058912** [comment]
@JessBoctor tried addressing the issue through [this commit](https://github.com/Automattic/wp-calypso/pull/78931/commits/a368e0d26e7b6255c34cd224f3d110799f2d0fe5).

File: client/my-sites/checkout/checkout-thank-you/style.scss (hunk 6 above)
Comment (candy02058912) [reply client/my-sites/checkout/checkout-thank-you/style.scss]: Moved! https://github.com/Automattic/wp-calypso/pull/78931/commits/552c7c7377dff0874bcfe01c400f6480521ab1e4

**Comment by candy02058912** [comment]
> Nice, that sounds good. Will it also lives inside redesign-v2 folder?

@paulopmt1 yup that's what I'm planning in https://github.com/Automattic/dotcom-forge/issues/2964

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8353915

Thank you @candy02058912 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8353915).
